<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>enrichment with other gene sets</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">multi_omics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/han16/multi_omics">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">enrichment with other gene sets</h1>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-10-15
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 4
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 3
</p>
<p>
<strong>Knit directory:</strong> <code>multi_omics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240123code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240123)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240123code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240123)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongdetected">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>Cache:</strong> detected </a>
</p>
</div>
<div id="strongCachestrongdetected" class="panel-collapse collapse">
<div class="panel-body">
The following chunks had caches available:
<ul>
<li>
session-info-chunk-inserted-by-workflowr
</li>
<li>
unnamed-chunk-10
</li>
<li>
unnamed-chunk-11
</li>
<li>
unnamed-chunk-12
</li>
<li>
unnamed-chunk-13
</li>
<li>
unnamed-chunk-14
</li>
<li>
unnamed-chunk-2
</li>
<li>
unnamed-chunk-3
</li>
<li>
unnamed-chunk-4
</li>
<li>
unnamed-chunk-5
</li>
<li>
unnamed-chunk-6
</li>
<li>
unnamed-chunk-7
</li>
<li>
unnamed-chunk-8
</li>
<li>
unnamed-chunk-9
</li>
</ul>
<p>To ensure reproducibility of the results, delete the cache directory
<code>part3_cache</code> and re-run the analysis. To have workflowr
automatically delete the cache directory prior to building the file, set
<code>delete_cache = TRUE</code> when running <code>wflow_build()</code>
or <code>wflow_publish()</code>.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it
difficult for you and others to run your code on a different machine.
Change the absolute path(s) below to the suggested relative path(s) to
make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/han/Projects/Multi_Omics/multi_omics/data/gene_pfc35_20240416.RData
</td>
<td style="text-align:left;">
data/gene_pfc35_20240416.RData
</td>
</tr>
<tr>
<td style="text-align:left;">
C:/han/Projects/Multi_Omics/multi_omics/data/dat_pfc35_20240328.RData
</td>
<td style="text-align:left;">
data/dat_pfc35_20240328.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomhan16multiomicstree446a37d5f664df42bcf06ef4dda28981df4659fctargetblank446a37da">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/han16/multi_omics/tree/446a37d5f664df42bcf06ef4dda28981df4659fc" target="_blank">446a37d</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomhan16multiomicstree446a37d5f664df42bcf06ef4dda28981df4659fctargetblank446a37da"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/han16/multi_omics/tree/446a37d5f664df42bcf06ef4dda28981df4659fc" target="_blank">446a37d</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    analysis/.Rhistory
    Ignored:    analysis/SCZ_gene_list_cache/
    Ignored:    analysis/part1_cache/
    Ignored:    analysis/part2_cache/
    Ignored:    analysis/part3_cache/
    Ignored:    analysis/part4_cache/
    Ignored:    analysis/part5_cache/
    Ignored:    analysis/part6_cache/

Unstaged changes:
    Modified:   analysis/part3.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/part3.Rmd</code>) and HTML
(<code>docs/part3.html</code>) files. If you’ve configured a remote Git
repository (see <code>?wflow_git_remote</code>), click on the hyperlinks
in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/bcc29e3af2865c112d1befed1a9416ab3115ad18/docs/part3.html" target="_blank">bcc29e3</a>
</td>
<td>
han
</td>
<td>
2024-08-21
</td>
<td>
8/21/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/1e0f797645aee1b17a729da8acc94753e226a2f0/analysis/part3.Rmd" target="_blank">1e0f797</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
<td>
6/24/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/1e0f797645aee1b17a729da8acc94753e226a2f0/docs/part3.html" target="_blank">1e0f797</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
<td>
6/24/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/346185ddd373193a94c40c02e1177aaae1e2b445/analysis/part3.Rmd" target="_blank">346185d</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
<td>
6/24/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/346185ddd373193a94c40c02e1177aaae1e2b445/docs/part3.html" target="_blank">346185d</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
<td>
6/24/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/6af4748783776a4be9f7f93ae66036c522b26b64/analysis/part3.Rmd" target="_blank">6af4748</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
<td>
6/24/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/6af4748783776a4be9f7f93ae66036c522b26b64/docs/part3.html" target="_blank">6af4748</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
<td>
6/24/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/08a59599d1ea69c570997989c078c06a9cfbd870/analysis/part3.Rmd" target="_blank">08a5959</a>
</td>
<td>
han
</td>
<td>
2024-06-12
</td>
<td>
6/12/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/50f9687e1faf522773dc93384ec5ce3d62e20eba/docs/part3.html" target="_blank">50f9687</a>
</td>
<td>
han
</td>
<td>
2024-06-10
</td>
<td>
6/10/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/0ed733af7033119e0705eb4db3eb25581044fbc1/analysis/part3.Rmd" target="_blank">0ed733a</a>
</td>
<td>
han
</td>
<td>
2024-05-14
</td>
<td>
5/14/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/0ed733af7033119e0705eb4db3eb25581044fbc1/docs/part3.html" target="_blank">0ed733a</a>
</td>
<td>
han
</td>
<td>
2024-05-14
</td>
<td>
5/14/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/1ab7c4d13e1d009450b46d704ef88a275e281b22/analysis/part3.Rmd" target="_blank">1ab7c4d</a>
</td>
<td>
han
</td>
<td>
2024-05-07
</td>
<td>
5/7/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/1ab7c4d13e1d009450b46d704ef88a275e281b22/docs/part3.html" target="_blank">1ab7c4d</a>
</td>
<td>
han
</td>
<td>
2024-05-07
</td>
<td>
5/7/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/4ab24910196b8ca376bf8aec640e3cf530bd7097/analysis/part3.Rmd" target="_blank">4ab2491</a>
</td>
<td>
han
</td>
<td>
2024-05-06
</td>
<td>
5/6/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/4ab24910196b8ca376bf8aec640e3cf530bd7097/docs/part3.html" target="_blank">4ab2491</a>
</td>
<td>
han
</td>
<td>
2024-05-06
</td>
<td>
5/6/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/52ff0c059e151bf40d7d7f5677082a2127691fe2/analysis/part3.Rmd" target="_blank">52ff0c0</a>
</td>
<td>
han
</td>
<td>
2024-05-06
</td>
<td>
5/6/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/52ff0c059e151bf40d7d7f5677082a2127691fe2/docs/part3.html" target="_blank">52ff0c0</a>
</td>
<td>
han
</td>
<td>
2024-05-06
</td>
<td>
5/6/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/d03399058323a650ce5e7033a354fad4e67da857/analysis/part3.Rmd" target="_blank">d033990</a>
</td>
<td>
han
</td>
<td>
2024-05-02
</td>
<td>
5/2/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/d03399058323a650ce5e7033a354fad4e67da857/docs/part3.html" target="_blank">d033990</a>
</td>
<td>
han
</td>
<td>
2024-05-02
</td>
<td>
5/2/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/e2afa26a573cbfefc93797d9a63cd6c9ac72cf95/analysis/part3.Rmd" target="_blank">e2afa26</a>
</td>
<td>
han
</td>
<td>
2024-04-29
</td>
<td>
4/29/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/e2afa26a573cbfefc93797d9a63cd6c9ac72cf95/docs/part3.html" target="_blank">e2afa26</a>
</td>
<td>
han
</td>
<td>
2024-04-29
</td>
<td>
4/29/2024
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/han16/multi_omics/blob/afb290c8c6568bb701909941259b4265610c2b1b/analysis/part3.Rmd" target="_blank">afb290c</a>
</td>
<td>
han
</td>
<td>
2024-04-26
</td>
<td>
4/26/2024
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/han16/multi_omics/afb290c8c6568bb701909941259b4265610c2b1b/docs/part3.html" target="_blank">afb290c</a>
</td>
<td>
han
</td>
<td>
2024-04-26
</td>
<td>
4/26/2024
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="venn-diagrams-of-genes-shared-across-different-disorders"
class="section level2">
<h2>Venn diagrams of genes shared across different disorders</h2>
<pre class="r"><code>load(&quot;C:/han/Projects/Multi_Omics/multi_omics/data/gene_pfc35_20240416.RData&quot;) # load gene_pfc35
load(&quot;C:/han/Projects/Multi_Omics/multi_omics/data/dat_pfc35_20240328.RData&quot;) # load data dat_pfc35
Fu_gene_list=as_tibble(read.csv(&quot;C:/han/Dataset/20231221_PMID_35982160/Fu 2022 Suppl-41588_2022_1104_MOESM3_ESM.csv&quot;))
ASC_gene=Fu_gene_list %&gt;% filter(ASD185==&quot;TRUE&quot;) %&gt;% dplyr::select(gene_gencodeV33) %&gt;% pull
neuro_disorers=SCHEMA_gene_results$`Table S11 - Genes from genome-w`
DD_ID_gene=as_tibble(neuro_disorers) %&gt;% filter(Category==&quot;DD/ID genes&quot;) %&gt;% dplyr::select(`Gene name`) %&gt;% pull()
SCZ_GWAS_gene=as_tibble(neuro_disorers) %&gt;% filter(Category==&quot;Schizophrenia GWAS genes&quot;) %&gt;% dplyr::select(`Gene name`) %&gt;% pull()

x=list(
  DAWN_gene=genes_pfc35,
  #Autism_gene=as_tibble(neuro_disorers) %&gt;% filter(Category==&quot;ASD genes&quot;) %&gt;% dplyr::select(`Gene name`) %&gt;% pull(),
  Autism_gene=ASC_gene,
  DD_ID_gene=DD_ID_gene,
  SCZ_GWAS_gene=SCZ_GWAS_gene

)
library(VennDiagram)

display_venn &lt;- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object &lt;- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}

#png(&quot;C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig_overlapping_genes.png&quot;)

display_venn(
        x,
    #    category.names = c(&quot;MIRAGE FDR genes&quot; , &quot;Autism genes&quot; , &quot;DD/ID genes&quot;, &quot;SCZ GWAS genes&quot;),
    category.names = c(&quot;DAWN genes&quot; , &quot;Autism genes&quot; , &quot;DD/ID genes&quot;, &quot;SCZ GWAS genes&quot;),
        # Circles
        lwd = 2,
        lty = &#39;blank&#39;,
     #   fill = c(&quot;#999999&quot;, &quot;#E69F00&quot;, &quot;#56B4E9&quot;, &quot;#009E73&quot;),
    fill = c(&quot;red&quot;, &quot;blue&quot;, &quot;green&quot;, &quot;yellow&quot;),
        # Numbers
        cex = 1.5,
        fontface = &quot;italic&quot;,
        # Set names
        cat.cex = 1.5,
        cat.fontface = &quot;bold&quot;,
        cat.default.pos = &quot;outer&quot;,
       # cat.dist = c(0.055, 0.055, 0.1, 0.1, 0.3 )
         cat.dist = c(0.055, 0.055, 0.1, 0.1 )
)</code></pre>
<p><img src="figure/part3.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-3-1">
Past versions of unnamed-chunk-3-1.png
</button>
</p>
<div id="fig-unnamed-chunk-3-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/multi_omics/blob/6af4748783776a4be9f7f93ae66036c522b26b64/docs/figure/part3.Rmd/unnamed-chunk-3-1.png" target="_blank">6af4748</a>
</td>
<td>
han
</td>
<td>
2024-06-24
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<div id="dawn-genes-overlpping-with-autism" class="section level3">
<h3>DAWN genes overlpping with autism</h3>
<div id="overlapping-genes-between-dawn-46-genes-and-asc-genes"
class="section level4">
<h4>overlapping genes between DAWN 46 genes and ASC genes</h4>
<pre class="r"><code>intersect(genes_pfc35, ASC_gene)</code></pre>
<pre><code>[1] &quot;ASH1L&quot;   &quot;CGREF1&quot;  &quot;FAM120A&quot; &quot;KDM6B&quot;   &quot;ZMYM2&quot;   &quot;ZMYND11&quot;</code></pre>
<pre class="r"><code>fisher.test(matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), ASC_gene)), length(intersect(genes_pfc35, ASC_gene))), nrow=2))</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), ASC_gene)), length(intersect(genes_pfc35, ASC_gene))), nrow = 2)
p-value = 0.004104
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  1.51063 10.79097
sample estimates:
odds ratio 
  4.438854 </code></pre>
</div>
<div id="overlapping-genes-between-schema-fdr-genes-and-asc-genes"
class="section level4">
<h4>overlapping genes between SCHEMA FDR genes and ASC genes</h4>
<pre class="r"><code>SCHEMA_FDR_gene=SCHEMA_gene_results$`Table S5 - Gene Results`[order(SCHEMA_gene_results$`Table S5 - Gene Results`$`P ca/co (comb)`, decreasing = F),]$`Gene Symbol`[1:32]
intersect(SCHEMA_FDR_gene, ASC_gene)  # overlapping genes between SCHEMA and 185 autism genes </code></pre>
<pre><code>[1] &quot;FAM120A&quot; &quot;ASH1L&quot;   &quot;ZMYM2&quot;   &quot;KDM6B&quot;  </code></pre>
</div>
<div id="new-overlapping-genes-between-dawn-46-genes-and-asc-genes"
class="section level4">
<h4>new overlapping genes between DAWN 46 genes and ASC genes</h4>
<pre class="r"><code>setdiff(intersect(genes_pfc35, ASC_gene),intersect(SCHEMA_FDR_gene, ASC_gene)) </code></pre>
<pre><code>[1] &quot;CGREF1&quot;  &quot;ZMYND11&quot;</code></pre>
</div>
</div>
<div id="dawn-genes-overlpping-with-ddid" class="section level3">
<h3>DAWN genes overlpping with DD/ID</h3>
<pre class="r"><code>intersect(genes_pfc35, DD_ID_gene)</code></pre>
<pre><code> [1] &quot;ASH1L&quot;   &quot;CACNA1G&quot; &quot;GRIN2A&quot;  &quot;KDM6B&quot;   &quot;NR4A2&quot;   &quot;SRRM2&quot;   &quot;STAG1&quot;  
 [8] &quot;TRIO&quot;    &quot;ZMYM2&quot;   &quot;ZMYND11&quot;</code></pre>
<pre class="r"><code>fisher.test(matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), DD_ID_gene)), length(intersect(genes_pfc35, DD_ID_gene))), nrow=2))</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), DD_ID_gene)), length(intersect(genes_pfc35, DD_ID_gene))), nrow = 2)
p-value = 8.931e-05
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  2.295333 10.786560
sample estimates:
odds ratio 
  5.218651 </code></pre>
<pre class="r"><code>intersect(intersect(genes_pfc35, DD_ID_gene), ASC_gene)</code></pre>
<pre><code>[1] &quot;ASH1L&quot;   &quot;KDM6B&quot;   &quot;ZMYM2&quot;   &quot;ZMYND11&quot;</code></pre>
<pre class="r"><code>intersect(intersect(genes_pfc35, DD_ID_gene), SCZ_GWAS_gene)</code></pre>
<pre><code>[1] &quot;GRIN2A&quot;</code></pre>
<ul>
<li>10 DD/ID genes shared with SCHEMA FDR genes: SETD1A, TRIO, CACNA1G,
GRIN2A, SRRM2, ZMYM2, KDM6B, ASH1L, STAG1, HIST1H1E, from SCHEMA Supp
table</li>
</ul>
<pre class="r"><code>DD_ID_SCHEMA=c(&quot;SETD1A&quot;, &quot;TRIO&quot;, &quot;CACNA1G&quot;, &quot;GRIN2A&quot;, &quot;SRRM2&quot;, &quot;ZMYM2&quot;, &quot;KDM6B&quot;, &quot;ASH1L&quot;, &quot;STAG1&quot;, &quot;HIST1H1E&quot;)
setdiff(intersect(genes_pfc35, DD_ID_gene), DD_ID_SCHEMA)</code></pre>
<pre><code>[1] &quot;NR4A2&quot;   &quot;ZMYND11&quot;</code></pre>
<ul>
<li>two new genes NR4A2, ZMYND11 overlapping with DD/ID, compared to
SCHEMA results.</li>
</ul>
</div>
<div id="dawn-genes-overlpping-with-scz-gwas-gene"
class="section level3">
<h3>DAWN genes overlpping with SCZ GWAS gene</h3>
<pre class="r"><code>intersect(genes_pfc35, SCZ_GWAS_gene)</code></pre>
<pre><code>[1] &quot;GRIN2A&quot; &quot;SP4&quot;   </code></pre>
<pre class="r"><code>fisher.test(matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), SCZ_GWAS_gene)), length(intersect(genes_pfc35, SCZ_GWAS_gene))), nrow=2))</code></pre>
<pre><code>
    Fisher&#39;s Exact Test for Count Data

data:  matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), SCZ_GWAS_gene)), length(intersect(genes_pfc35, SCZ_GWAS_gene))), nrow = 2)
p-value = 0.03269
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.8624537 34.9613068
sample estimates:
odds ratio 
  7.913397 </code></pre>
</div>
</div>
<div id="or-of-dawn-genes-in-different-gene-sets"
class="section level2">
<h2>OR of DAWN genes in different gene sets</h2>
<pre class="r"><code># enrichment analysis
all_gene=colnames(dat_pfc35)
Autism_gene=ASC_gene
DD_ID_gene=as_tibble(neuro_disorers) %&gt;% filter(Category==&quot;DD/ID genes&quot;) %&gt;% dplyr::select(`Gene name`) %&gt;% pull()
SCZ_GWAS_gene=as_tibble(neuro_disorers) %&gt;% filter(Category==&quot;Schizophrenia GWAS genes&quot;) %&gt;% dplyr::select(`Gene name`) %&gt;% pull()
SCHEMA_FDR_gene=SCHEMA_gene_results$`Table S5 - Gene Results`[order(SCHEMA_gene_results$`Table S5 - Gene Results`$`P ca/co (comb)`, decreasing = F),]$`Gene Symbol`[1:32]
dawn_new_gene=setdiff(genes_pfc35, SCHEMA_FDR_gene)

gene_set=list(all_gene=all_gene, Autism_gene=Autism_gene, DD_ID_gene=DD_ID_gene, SCZ_GWAS_gene=SCZ_GWAS_gene, SCHEMA_FDR_gene=SCHEMA_FDR_gene, DAWN_gene=genes_pfc35, DAWN_new_gene=dawn_new_gene)

##########################
OR=numeric(); pvalue_dawn=numeric(); pvalue_dawn_new=numeric(); pvalue_schema=numeric()
percentage=matrix(nrow=4, ncol=3)
rownames(percentage)=c(&quot;dawn_gene&quot;, &quot;all_gene&quot;, &quot;schema_fdr&quot;, &quot;dawn_new_gene&quot;)
colnames(percentage)=names(gene_set)[2:4]
DAWN_overlap_matrix=list()
DAWN_new_overlap_matrix=list()
for (i in 2:4)
{
dawn_matrix=matrix(c(length(intersect(gene_set$DAWN_gene, gene_set[[i]])), 
                                 length(intersect(gene_set$all_gene, gene_set[[i]])), 
                                 length(gene_set$DAWN_gene), 
                                 length(gene_set$all_gene)), nrow=2)
  fisher_test=fisher.test(dawn_matrix)
  OR[i-1]=fisher_test$estimate; pvalue_dawn[i-1]=fisher_test$p.value; 
  DAWN_overlap_matrix[[i-1]]=dawn_matrix
  
  dawn_new_matrix=matrix(c(length(intersect(gene_set$DAWN_new_gene, gene_set[[i]])), 
                                        length(intersect(gene_set$all_gene, gene_set[[i]])), 
                                        length(gene_set$DAWN_new_gene), 
                                        length(gene_set$all_gene)), nrow=2)
pvalue_dawn_new[i-1]=fisher.test(dawn_new_matrix)$p.value
DAWN_new_overlap_matrix[[i-1]]=dawn_new_matrix
  
 percentage[1,(i-1)]=length(intersect(gene_set$DAWN_gene, gene_set[[i]]))/length(gene_set$DAWN_gene)
 percentage[2,(i-1)]=length(intersect(gene_set$all_gene, gene_set[[i]]))/length(gene_set$all_gene)
 percentage[3,(i-1)]=length(intersect(gene_set$SCHEMA_FDR_gene, gene_set[[i]]))/length(gene_set$SCHEMA_FDR_gene)
 percentage[4,(i-1)]=length(intersect(gene_set$DAWN_new_gene, gene_set[[i]]))/length(gene_set$DAWN_new_gene)
                                                                                                                   
 pvalue_schema[i-1]=fisher.test(matrix(c(length(intersect(gene_set$SCHEMA_FDR_gene, gene_set[[i]])), 
                                 length(intersect(gene_set$all_gene, gene_set[[i]])), 
                                 length(gene_set$SCHEMA_FDR_gene), 
                                 length(gene_set$all_gene)), nrow=2))$p.value
 
}</code></pre>
<p><img src="figure/part3.Rmd/unnamed-chunk-11-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-11-1">
Past versions of unnamed-chunk-11-1.png
</button>
</p>
<div id="fig-unnamed-chunk-11-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/multi_omics/blob/0ed733af7033119e0705eb4db3eb25581044fbc1/docs/figure/part3.Rmd/unnamed-chunk-11-1.png" target="_blank">0ed733a</a>
</td>
<td>
han
</td>
<td>
2024-05-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/han16/multi_omics/blob/1ab7c4d13e1d009450b46d704ef88a275e281b22/docs/figure/part3.Rmd/unnamed-chunk-11-1.png" target="_blank">1ab7c4d</a>
</td>
<td>
han
</td>
<td>
2024-05-07
</td>
</tr>
</tbody>
</table>
</div>
</div>
<ul>
<li>numbers over bars are p values.</li>
</ul>
</div>
<div id="enrichment-of-dawn-new-genes-with-other-gene-sets"
class="section level2">
<h2>enrichment of DAWN (new) genes with other gene sets</h2>
<pre class="r"><code>percentage_plot=data.frame(prop=c(percentage[2,],percentage[1,], percentage[3,]), genes=rep(c(&quot;Autism genes&quot;, &quot;DD/ID genes&quot;,    &quot;SCZ GWAS genes&quot;),3), group=rep(c(&quot;all genes&quot;, &quot;DAWN genes&quot;, &quot;SCHEMA FDR genes&quot;), each=3))

textsize=5
#                                                                                                 par(mar=rep(0,4))
topmirage_enrichment=
  ggplot(percentage_plot, aes(genes, prop)) + 
   geom_bar(aes(fill = group), stat = &quot;identity&quot;, position = &quot;dodge&quot;)+
   theme_classic()+
   xlab(&quot;&quot;)+ylab(&quot;Proportion of overlapping genes in the gene set&quot;)+
  theme(axis.text.y = element_text(size = 15))+ # this is for numbers on y axis 
   ylim(c(0, 0.35))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face=&quot;bold&quot;))+
   ggtitle(&quot;Enrichment of DAWN genes with other gene sets&quot;)+
    theme(legend.title = element_blank())+
   geom_signif(annotations =&quot;0.004&quot;, y_position = 0.14  ,xmin=0.7, xmax=1, size=1, vjust=-0.5, textsize=textsize)+ # textsize is for text over the bar; size is for width of lines 
  geom_signif(annotations =&quot;0.020&quot;, y_position = 0.18  ,xmin=0.7, xmax=1.3, size=1, vjust=-0.5, textsize=textsize)+ # textsize is for text over the bar; size is for width of lines
     geom_signif(annotations =&quot;6e-06&quot;, y_position = 0.325  ,xmin=1.7, xmax=2.3, size=1, vjust=-0.5, textsize=textsize)+
  geom_signif(annotations =&quot;8e-05&quot;, y_position = 0.23  ,xmin=1.7, xmax=2, size=1, vjust=-0.5, textsize=textsize)+
   geom_signif(annotations =&quot;0.002&quot;, y_position = 0.09  ,xmin=2.7, xmax=3.3, size=1, vjust=-0.5, textsize=textsize)+
  geom_signif(annotations =&quot;0.033&quot;, y_position = 0.052  ,xmin=2.7, xmax=3.0, size=1, vjust=-0.5, textsize=textsize)+
   theme(legend.position=&quot;bottom&quot;)+
   theme(axis.text.x = element_text(angle = 0, vjust = 0.5, size=15))+
 theme(text = element_text(size=20))+ # size is for labels on x axis and legend title 
  theme(axis.title = element_text(size = 13))+ # Font Size of Axis Titles
 theme(legend.spacing.x = unit(1.0, &#39;cm&#39;)) # add space between symbols in legend 
###################### compute p value 

#png(&quot;C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig-enrichment_of_top_genes.png&quot;)
topmirage_enrichment</code></pre>
<p><img src="figure/part3.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-12-1">
Past versions of unnamed-chunk-12-1.png
</button>
</p>
<div id="fig-unnamed-chunk-12-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/multi_omics/blob/0ed733af7033119e0705eb4db3eb25581044fbc1/docs/figure/part3.Rmd/unnamed-chunk-12-1.png" target="_blank">0ed733a</a>
</td>
<td>
han
</td>
<td>
2024-05-14
</td>
</tr>
<tr>
<td>
<a href="https://github.com/han16/multi_omics/blob/52ff0c059e151bf40d7d7f5677082a2127691fe2/docs/figure/part3.Rmd/unnamed-chunk-12-1.png" target="_blank">52ff0c0</a>
</td>
<td>
han
</td>
<td>
2024-05-06
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()</code></pre>
<pre class="r"><code>percentage_plot=data.frame(prop=c(percentage[2,], percentage[1,], percentage[4,]), genes=rep(c(&quot;Autism genes&quot;, &quot;DD/ID genes&quot;, &quot;SCZ GWAS genes&quot;),3), group=rep(c(&quot;all genes&quot;, &quot;DAWN genes&quot;, &quot;DAWN new genes&quot;), each=3))
topmirage_enrichment=
  ggplot(percentage_plot, aes(genes, prop)) + 
   geom_bar(aes(fill = group), stat = &quot;identity&quot;, position = &quot;dodge&quot;)+
   theme_classic()+
   xlab(&quot;&quot;)+ylab(&quot;Proportion of genes in the gene set&quot;)+
  theme(axis.text.y = element_text(size = 13))+ # this is for numbers on y axis 
   ylim(c(0, 0.35))+
   theme(plot.title = element_text(hjust = 0.5, size=15, face=&quot;bold&quot;))+
   ggtitle(&quot;Enrichment of DAWN genes with other gene sets&quot;)+
    theme(legend.title = element_blank())+
   geom_signif(annotations =&quot;0.113&quot;, y_position = 0.175  ,xmin=0.7, xmax=1.3, size=1, vjust=-0.5, textsize=5.0)+ # textsize is for text over the bar; size is for width of lines 
  geom_signif(annotations =&quot;0.004&quot;, y_position = 0.135  ,xmin=0.7, xmax=1, size=1, vjust=-0.5, textsize=5.0)+ # textsize is for text over the bar; size is for width of lines
     geom_signif(annotations =&quot;8e-05&quot;, y_position = 0.225  ,xmin=1.7, xmax=2, size=1, vjust=-0.5, textsize=5.0)+
    geom_signif(annotations =&quot;0.541&quot;, y_position = 0.26  ,xmin=1.7, xmax=2.3, size=1, vjust=-0.5, textsize=5.0)+
   geom_signif(annotations =&quot;0.033&quot;, y_position = 0.05  ,xmin=2.7, xmax=3, size=1, vjust=-0.5, textsize=5.0)+
     geom_signif(annotations =&quot;1.000&quot;, y_position = 0.09  ,xmin=2.7, xmax=3.3, size=1, vjust=-0.5, textsize=5.0)+
   theme(legend.position=&quot;bottom&quot;)+
   theme(axis.text.x = element_text(angle = 0, vjust = 0.5, size=15))+
 theme(text = element_text(size=15))+ # size is for labels on x axis 
  theme(axis.title = element_text(size = 15))+ # Font Size of Axis Titles
 theme(legend.spacing.x = unit(0.5, &#39;cm&#39;)) # add space between symbols in legend 

#png(&quot;C:/Users/hans/OneDrive - Marquette University/SCHEMA/Fig-enrichment_of_top_genes.png&quot;)
topmirage_enrichment</code></pre>
<p><img src="figure/part3.Rmd/unnamed-chunk-14-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-14-1">
Past versions of unnamed-chunk-14-1.png
</button>
</p>
<div id="fig-unnamed-chunk-14-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/han16/multi_omics/blob/0ed733af7033119e0705eb4db3eb25581044fbc1/docs/figure/part3.Rmd/unnamed-chunk-14-1.png" target="_blank">0ed733a</a>
</td>
<td>
han
</td>
<td>
2024-05-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>#dev.off()
###########  p values are pvalue_dawn; pvalue_dawn_new</code></pre>
</div>
<div id="overlap-with-mirage-fdr-genes" class="section level2">
<h2>overlap with MIRAGE FDR genes</h2>
<pre class="r"><code>##############
##########
rm(&quot;BF.PP.gene&quot;)
#BF.PP.gene=as_tibble(read.csv(&quot;C:\\han\\Dataset\\SCHEMA\\MIRAGE_on_meta_sample_estimate_delta_3vargroups_20231027.csv&quot;, header=T))
load(&quot;C:\\han\\Dataset\\SCHEMA\\MIRAGE_on_meta_sample_estimate_delta_2vargroups_for_constraint_gene_20231120.RData&quot;)

fdr_function=function(fdr_threshold, data)
{
tau=seq(0, 0.999, by=0.001)
num_pred=NULL
false_disc=NULL
FDR=NULL
for (i in 1:length(tau))
{
num_pred[i]=sum(ifelse(data$post_prob&gt;tau[i], 1, 0))
false_disc[i]=sum((1-data$post_prob)*ifelse(data$post_prob&gt;tau[i], 1, 0))
FDR[i]=false_disc[i]/num_pred[i]
}
tau_fdr=tibble(tau=tau, fdr=FDR)%&gt;%drop_na()  # drop rows with NA
tau_at_threshold=tau_fdr%&gt;%filter(fdr&lt;=fdr_threshold)
if (nrow(tau_at_threshold)==0)
  return(&quot;No gene is selected&quot;)
if (nrow(tau_at_threshold)&gt;0)
{
  tau_threshold=tau_at_threshold$tau[1]
  return(data%&gt;% filter(post_prob&gt;tau_threshold))
}

}</code></pre>
<pre class="r"><code>input_data=BF.PP.gene
colnames(input_data)[3]=&quot;post_prob&quot;
FDR_gene=fdr_function(fdr_threshold = 0.05, input_data)</code></pre>
<pre class="r"><code>intersect(FDR_gene$Gene, genes_pfc35)  # overlapping genes between 110 MIRAGE FDR genes and 46 DAWN genes</code></pre>
<pre><code> [1] &quot;AKAP11&quot;   &quot;TRAPPC10&quot; &quot;ANKRD12&quot;  &quot;ZMYM2&quot;    &quot;HERC1&quot;    &quot;HCN4&quot;    
 [7] &quot;SP4&quot;      &quot;RB1CC1&quot;   &quot;SOBP&quot;     &quot;STAG1&quot;    &quot;GRIA3&quot;    &quot;DAGLA&quot;   
[13] &quot;SCAF1&quot;    &quot;HMGCR&quot;    &quot;XPO7&quot;     &quot;GRIN2A&quot;   &quot;FAM120A&quot;  &quot;ZMYND11&quot; 
[19] &quot;KDM6B&quot;    &quot;CUL1&quot;     &quot;TRIO&quot;     &quot;FYN&quot;      &quot;ASH1L&quot;    &quot;CACNA2D1&quot;
[25] &quot;SRRM2&quot;    &quot;MAGI2&quot;    &quot;CACNA1G&quot; </code></pre>
<pre class="r"><code>setdiff(intersect(FDR_gene$Gene, genes_pfc35), SCHEMA_FDR_gene)  # 18 new DAWN genes overlapping with 110 MIRAGE FDR genes </code></pre>
<pre><code>[1] &quot;TRAPPC10&quot; &quot;SOBP&quot;     &quot;SCAF1&quot;    &quot;HMGCR&quot;    &quot;ZMYND11&quot;  &quot;FYN&quot;      &quot;CACNA2D1&quot;</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.4.0 (2024-04-24 ucrt)
Platform: x86_64-w64-mingw32/x64
Running under: Windows 10 x64 (build 19045)

Matrix products: default


locale:
[1] LC_COLLATE=English_United States.utf8 
[2] LC_CTYPE=English_United States.utf8   
[3] LC_MONETARY=English_United States.utf8
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.utf8    

time zone: America/Chicago
tzcode source: internal

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] VennDiagram_1.7.3   futile.logger_1.4.3 ggpubr_0.6.0       
 [4] gridExtra_2.3       lubridate_1.9.3     forcats_1.0.0      
 [7] stringr_1.5.1       purrr_1.0.2         readr_2.1.5        
[10] tidyr_1.3.1         tibble_3.2.1        ggplot2_3.5.1      
[13] tidyverse_2.0.0     dplyr_1.1.4         RSQLite_2.3.6      
[16] knitr_1.46         

loaded via a namespace (and not attached):
 [1] gtable_0.3.5         xfun_0.43            bslib_0.7.0         
 [4] rstatix_0.7.2        tzdb_0.4.0           vctrs_0.6.5         
 [7] tools_4.4.0          generics_0.1.3       fansi_1.0.6         
[10] highr_0.10           blob_1.2.4           pkgconfig_2.0.3     
[13] readxl_1.4.3         lifecycle_1.0.4      farver_2.1.2        
[16] compiler_4.4.0       git2r_0.33.0         munsell_0.5.1       
[19] carData_3.0-5        httpuv_1.6.15        htmltools_0.5.8.1   
[22] sass_0.4.9           yaml_2.3.8           later_1.3.2         
[25] pillar_1.9.0         car_3.1-2            jquerylib_0.1.4     
[28] whisker_0.4.1        cachem_1.0.8         abind_1.4-5         
[31] tidyselect_1.2.1     digest_0.6.35        stringi_1.8.4       
[34] labeling_0.4.3       rprojroot_2.0.4      fastmap_1.1.1       
[37] colorspace_2.1-0     cli_3.6.2            magrittr_2.0.3      
[40] utf8_1.2.4           broom_1.0.5          withr_3.0.0         
[43] scales_1.3.0         promises_1.3.0       backports_1.4.1     
[46] bit64_4.0.5          timechange_0.3.0     lambda.r_1.2.4      
[49] rmarkdown_2.26       bit_4.0.5            cellranger_1.1.0    
[52] ggsignif_0.6.4       workflowr_1.7.1      hms_1.1.3           
[55] memoise_2.0.1        evaluate_0.23        futile.options_1.0.1
[58] rlang_1.1.3          Rcpp_1.0.12          glue_1.7.0          
[61] DBI_1.2.3            formatR_1.14         rstudioapi_0.16.0   
[64] jsonlite_1.8.8       R6_2.5.1             fs_1.6.4            </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
