---
title: "coexpression analysis by WGCNA"
output: html_document
date: "2024-04-26"
---

```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
knitr::opts_chunk$set(autodep = TRUE)
library(knitr)
library(RSQLite)
#install.packages("vctrs")
library(vctrs)
library(dplyr)
library(knitr)
knitr::opts_chunk$set(cache = TRUE, warning = FALSE, 
                      message = FALSE, cache.lazy = FALSE) # avoid error of long vectors https://stackoverflow.com/questions/39417003/long-vectors-not-supported-yet-error-in-rmd-but-not-in-r-script 
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
#library(DT)

#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
# BiocManager::install("GO.db")
library(GO.db)

#install.packages('BiocManager')
library(BiocManager)

#install.packages(c("matrixStats", "Hmisc", "splines", "foreach", "doParallel", "fastcluster", "dynamicTreeCut", "survival"))

#install.packages("htmltools")
#remotes::install_github("rstudio/htmltools")
library(htmltools)

#install package impute 
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("impute")
library(impute)

# install package preprocessCore
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("preprocessCore")
library(preprocessCore)

library(WGCNA)


#install_github("jdstorey/qvalue")   # install qvalue package 
#library(qvalue)
#library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
library(igraph)
```






```{r, echo=F, message=F, warning=F, results=F}
######## replicating the results 
###https://github.com/linnykos/covarianceSelection
# install dependent packages 

#install.packages("doMC", repos="http://R-Forge.R-project.org")
library("doMC")

# if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("org.Hs.eg.db")
library("org.Hs.eg.db") # Genome wide annotation for Human https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html

#download gz file to local folder and install https://cran.r-project.org/src/contrib/Archive/dequer/
library(dequer)


library("devtools")
devtools::install_github("linnylin92/covarianceSelection", subdir = "covarianceSelection")

```

## gene co-expression by WGCNA  

* [reference1](https://fuzzyatelin.github.io/bioanth-stats/module-F21-Group1/module-F21-Group1.html)
* [reference2](https://rstudio-pubs-static.s3.amazonaws.com/687551_ed469310d8ea4652991a2e850b0018de.html)
* [reference3](https://alexslemonade.github.io/refinebio-examples/04-advanced-topics/network-analysis_rnaseq_01_wgcna.html)


```{r, echo=F, message=F, warning=F}
#load("../data/dat_pfc35_20240328.RData")
load("C:/han/Projects/Multi_Omics/multi_omics/data/dat_pfc35_20240328.RData")
dat_pfc35_constraint_gene=dat_pfc35[,which(colnames(dat_pfc35) %in% scz$Gene)] # extract gene expression for 3100 genes at PFC
gsg=goodSamplesGenes(dat_pfc35_constraint_gene) # identify outlier genes 
summary(gsg) # no outlier genes 
```

```{r, echo=F, message=F, warning=F}
### identify outlier samples https://fuzzyatelin.github.io/bioanth-stats/module-F21-Group1/module-F21-Group1.html#Identifying_outlier_samples
sampleTree <- hclust(dist(dat_pfc35_constraint_gene), method = "average") #Clustering samples based on distance 

#Setting the graphical parameters
par(cex = 0.6);
par(mar = c(0,4,2,0))

#Plotting the cluster dendrogram
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)



par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
#draw on line to show cutoff height
abline(h = 40, col = "red")


cut.sampleTree <- cutreeStatic(sampleTree, cutHeight = 40, minSize = 10) #returns numeric vector
#Remove outlier
expression.data <- dat_pfc35_constraint_gene[cut.sampleTree==1, ]
dim(expression.data)
```
```{r, echo=F, message=F, warning=F}
spt <- pickSoftThreshold(expression.data)
#spt

#par(mar=c(1,1,1,1))
plot(spt$fitIndices[,1],spt$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"))
text(spt$fitIndices[,1],spt$fitIndices[,2],col="red")
abline(h=0.80,col="red")

#par(mar=c(1,1,1,1))
plot(spt$fitIndices[,1], spt$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(spt$fitIndices[,1], spt$fitIndices[,5], labels= spt$fitIndices[,1],col="red")
```

* choose $\beta$ that maximizes $R^2$, and minimizes mean connectivity 



```{r, echo=F, message=F, warning=F}
softPower <- 8
adjacency <- adjacency(expression.data, power = softPower)
#dim(adjacency)
threshold=0.5
adjacency_strong=matrix(nrow=nrow(adjacency), ncol=ncol(adjacency))
k=0; row_sub=numeric()
for(i in 1:nrow(adjacency))
{
  for (j in 1:ncol(adjacency))
  {
    if (abs(adjacency[i,j])<=threshold)
         adjacency_strong[i,j]=0
     if (abs(adjacency[i,j])>threshold)
    adjacency_strong[i,j]=1
  } 
  if (sum(adjacency_strong[i,])==1)
  {
    k=k+1
    row_sub[k]=i
  }
}  # end of for(i in 1:nrow(adjacency))
library(statnet)
adjacency_network=adjacency_strong[-row_sub, -row_sub]
Gene_selected=colnames(adjacency)[-row_sub]
rownames(adjacency_network)=Gene_selected
colnames(adjacency_network)=Gene_selected
dim(adjacency_network)
net=network(adjacency_network, directed=FALSE) # https://web.stanford.edu/class/bios221/labs/networks/lab_7_networks.html
nodeColors<-ifelse(Gene_selected %in% Gene_selected,"red","red")
plot(net, label=Gene_selected,vertex.col=nodeColors, vertex.size=10, label.cex=0.5) # https://statnet.org/workshop-intro-sna-tools/



#### overlap between 32 FDR genes and co-expressed genes 
overlap_gene=intersect(scz[scz$qval.meta<0.05,]$Gene, Gene_selected)


nodeColors<-ifelse(Gene_selected%in%overlap_gene,"yellow","red")
plot(net,label=Gene_selected,vertex.col=nodeColors,  vertex.size=10, label.cex=0.5)

#install.packages("statnet")
#library(statnet)
#heatmap(adjacency)
#plot(graph_from_adjacency_matrix(adjacency))
```




```{r, echo=F, message=F, warning=F}
#https://fuzzyatelin.github.io/bioanth-stats/module-F21-Group1/module-F21-Group1.html#Network_Construction
TOM <- TOMsimilarity(adjacency)
TOM.dissimilarity <- 1-TOM
#creating the dendrogram 
geneTree <- hclust(as.dist(TOM.dissimilarity), method = "average")
#plotting the dendrogram
#sizeGrWindow(12,9)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity", 
labels = FALSE, hang = 0.04)


Modules <- cutreeDynamic(dendro = geneTree, distM = TOM.dissimilarity, deepSplit = 2, pamRespectsDendro = FALSE, minClusterSize = 30)
table(Modules) #returns a table of the counts of factor levels in an object. In this case how many genes are assigned to each created module.

ModuleColors <- labels2colors(Modules) #assigns each module number a color
table(ModuleColors) #returns the counts for each color (aka the number of genes within each module)

#plots the gene dendrogram with the module colors
plotDendroAndColors(geneTree, ModuleColors,"Module",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05,
main = "Gene dendrogram and module colors")
```