---
title: "enrichment with other gene sets"
output:
  workflowr::wflow_html:
    code_folding: hide
    toc: true
---


```{r, echo=F, message=F, warning=F}
rm(list=ls())
set.seed(123)
knitr::opts_chunk$set(autodep = TRUE)
library(knitr)
library(RSQLite)
library(dplyr)
library(knitr)
library(tidyverse)
library(gridExtra)
library(ggpubr)
```



```{r, echo=F, message=F, warning=F, results=F, eval=T}
multiplesheets <- function(fname) {
   
  # getting info about all excel sheets
  sheets <- readxl::excel_sheets(fname)
  tibble <- lapply(sheets, function(x) readxl::read_excel(fname, sheet = x))
  data_frame <- lapply(tibble, as.data.frame)
    
  # assigning names to data frames
  names(data_frame) <- sheets
    
  # print data frame
  print(data_frame)
}

####
# load SCZ gene 
SCHEMA_gene_results=multiplesheets("C:\\han\\Dataset\\SCHEMA\\41586_2022_4556_MOESM3_ESM.xlsx")
scz1=data.frame(Gene=SCHEMA_gene_results$`Table S5 - Gene Results`$`Gene Symbol`, pval.meta=SCHEMA_gene_results$`Table S5 - Gene Results`$`P meta`, qval.meta=SCHEMA_gene_results$`Table S5 - Gene Results`$`Q meta`, dn_PTV=SCHEMA_gene_results$`Table S5 - Gene Results`$`De novo PTV`) 
scz1$dn_PTV[is.na(scz1$dn_PTV)]=0   # fill missing PTV with 0 
scz1=scz1%>% drop_na()  # remove NA's  ######## this is important 

scz2=scz1[-c(9934,  9935,  9936, 13100),]   #  remove duplicates which(duplicated(scz1$Gene)==T)
#unique_gene= unique(scz1$Gene)
#scz=data.frame(Gene=unique_gene, pval.meta=scz1$pval.meta[which(scz1$Gene %in% unique_gene)])  # remove duplicates 
```


## Venn diagrams of genes shared across different disorders 


```{r, echo=F, message=F, warning=F}
load("C:/han/Projects/Multi_Omics/multi_omics/data/gene_pfc35_20240416.RData") # load gene_pfc35
load("C:/han/Projects/Multi_Omics/multi_omics/data/dat_pfc35_20240328.RData") # load data dat_pfc35
Fu_gene_list=as_tibble(read.csv("C:/han/Dataset/20231221_PMID_35982160/Fu 2022 Suppl-41588_2022_1104_MOESM3_ESM.csv"))
ASC_gene=Fu_gene_list %>% filter(ASD185=="TRUE") %>% dplyr::select(gene_gencodeV33) %>% pull
neuro_disorers=SCHEMA_gene_results$`Table S11 - Genes from genome-w`
DD_ID_gene=as_tibble(neuro_disorers) %>% filter(Category=="DD/ID genes") %>% dplyr::select(`Gene name`) %>% pull()
SCZ_GWAS_gene=as_tibble(neuro_disorers) %>% filter(Category=="Schizophrenia GWAS genes") %>% dplyr::select(`Gene name`) %>% pull()

x=list(
  DAWN_gene=genes_pfc35,
  #Autism_gene=as_tibble(neuro_disorers) %>% filter(Category=="ASD genes") %>% dplyr::select(`Gene name`) %>% pull(),
  Autism_gene=ASC_gene,
  DD_ID_gene=DD_ID_gene,
  SCZ_GWAS_gene=SCZ_GWAS_gene

)
library(VennDiagram)

display_venn <- function(x, ...){
  library(VennDiagram)
  grid.newpage()
  venn_object <- venn.diagram(x, filename = NULL, ...)
  grid.draw(venn_object)
}

#png("C:\\Users\\hans\\OneDrive - Marquette University\\SCZ-DAWN\\Figures\\Fig_overlapping_genes.png")

display_venn(
        x,
    #    category.names = c("MIRAGE FDR genes" , "Autism genes" , "DD/ID genes", "SCZ GWAS genes"),
    category.names = c("DAWN genes" , "Autism genes" , "DD/ID genes", "SCZ GWAS genes"),
        # Circles
        lwd = 2,
        lty = 'blank',
     #   fill = c("#999999", "#E69F00", "#56B4E9", "#009E73"),
    fill = c("red", "blue", "green", "yellow"),
        # Numbers
        cex = 1.5,
        fontface = "italic",
        # Set names
        cat.cex = 1.5,
        cat.fontface = "bold",
        cat.default.pos = "outer",
       # cat.dist = c(0.055, 0.055, 0.1, 0.1, 0.3 )
         cat.dist = c(0.055, 0.055, 0.1, 0.1 )
)

#dev.off()
```




### DAWN genes overlpping with autism 


#### overlapping genes between DAWN 46 genes and ASC genes  


```{r,message=F, warning=F}
intersect(genes_pfc35, ASC_gene)
fisher.test(matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), ASC_gene)), length(intersect(genes_pfc35, ASC_gene))), nrow=2))
```
#### overlapping genes between SCHEMA FDR genes and ASC genes 


```{r, message=F, warning=F}
SCHEMA_FDR_gene=SCHEMA_gene_results$`Table S5 - Gene Results`[order(SCHEMA_gene_results$`Table S5 - Gene Results`$`P ca/co (comb)`, decreasing = F),]$`Gene Symbol`[1:32]
intersect(SCHEMA_FDR_gene, ASC_gene)  # overlapping genes between SCHEMA and 185 autism genes 
```

#### new overlapping genes between DAWN 46 genes and ASC genes 

```{r, message=F, warning=F}
setdiff(intersect(genes_pfc35, ASC_gene),intersect(SCHEMA_FDR_gene, ASC_gene)) 
```



### DAWN genes overlpping with DD/ID 

```{r, message=F, warning=F}
intersect(genes_pfc35, DD_ID_gene)
fisher.test(matrix(c(3100, 46, length(intersect(colnames(dat_pfc35), DD_ID_gene)), length(intersect(genes_pfc35, DD_ID_gene))), nrow=2))
intersect(intersect(genes_pfc35, DD_ID_gene), ASC_gene)
intersect(intersect(genes_pfc35, DD_ID_gene), SCZ_GWAS_gene)
```

* 10 DD/ID genes shared with SCHEMA FDR genes: SETD1A, TRIO, CACNA1G, GRIN2A, SRRM2, ZMYM2, KDM6B, ASH1L, STAG1, HIST1H1E, from SCHEMA Supp table 

```{r, message=F, warning=F}
DD_ID_SCHEMA=c("SETD1A", "TRIO", "CACNA1G", "GRIN2A", "SRRM2", "ZMYM2", "KDM6B", "ASH1L", "STAG1", "HIST1H1E")
setdiff(intersect(genes_pfc35, DD_ID_gene), DD_ID_SCHEMA)
```
