---
title: "use PFC35 data"
output: html_document
date: "2024-01-29"
---


```{r, echo=F, warning=F, message=F}
rm(list=ls())
set.seed(123)
knitr::opts_chunk$set(autodep = TRUE)
library(knitr)
library(RSQLite)
#install.packages("vctrs")
library(vctrs)
library(dplyr)
library(knitr)
knitr::opts_chunk$set(cache = TRUE, warning = FALSE, 
                      message = FALSE, cache.lazy = FALSE) # avoid error of long vectors https://stackoverflow.com/questions/39417003/long-vectors-not-supported-yet-error-in-rmd-but-not-in-r-script 
library(kableExtra)
library(RColorBrewer)
library(gplots)
library(tidyverse)
library(gridExtra)
library(ggpubr)
#library(DT)

#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")
# BiocManager::install("GO.db")
library(GO.db)

#install.packages('BiocManager')
library(BiocManager)

#install.packages(c("matrixStats", "Hmisc", "splines", "foreach", "doParallel", "fastcluster", "dynamicTreeCut", "survival"))

#install.packages("htmltools")
#remotes::install_github("rstudio/htmltools")
library(htmltools)

#install package impute 
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("impute")
library(impute)

# install package preprocessCore
#if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("preprocessCore")
library(preprocessCore)

library(WGCNA)


#install_github("jdstorey/qvalue")   # install qvalue package 
#library(qvalue)
#library("devtools")
#install.packages("Rtools")
#install_github('xinhe-lab/mirage')
library(mirage)
library(igraph)
```






```{r, echo=F, message=F, warning=F, results=F}
######## replicating the results 
###https://github.com/linnykos/covarianceSelection
# install dependent packages 

#install.packages("doMC", repos="http://R-Forge.R-project.org")
library("doMC")

# if (!require("BiocManager", quietly = TRUE))
#    install.packages("BiocManager")

#BiocManager::install("org.Hs.eg.db")
library("org.Hs.eg.db") # Genome wide annotation for Human https://bioconductor.org/packages/release/data/annotation/html/org.Hs.eg.db.html

#download gz file to local folder and install https://cran.r-project.org/src/contrib/Archive/dequer/
library(dequer)


library("devtools")
devtools::install_github("linnylin92/covarianceSelection", subdir = "covarianceSelection")

```




```{r, echo=F, message=F, warning=F, eval=F}
# step0_loading.R  https://github.com/linnykos/covarianceSelection/blob/master/main/step0_loading.R
# step0_loading.R loads the BrainSpan dataset and TADA dataset (i.e., matching the genes in both datasets, resolving gene synonyms, removing genes not expressed in the brain).

#if(verbose) print("Start of step 0: Loading")

#format the brainspan dataset
#load("../../raw_data/newGenexp.RData")
load("C:/han/Dataset/20231006_PMID_33731968/NIHMS1598069-supplement-Supp_1/newGenexp.RData")
rownames(genexp) <- genexp[,1]
genexp <- genexp[,-1]
genexp <- t(genexp)
genexp <- as.data.frame(genexp) # 1340 x 16947 ? what does row names indicate? sample information is in covarianceSelection::brainspan_id
#more data information: https://github.com/linnykos/covarianceSelection/blob/master/covarianceSelection/R/data.R  

#determine brain-expressed genes
brain_expression <- covarianceSelection::brain_expression # how these genes are known to be expressed genes???????? a data from Bernie Devlin   https://github.com/linnykos/covarianceSelection/blob/master/covarianceSelection/R/data.R
brain_genes <- brain_expression$Gene[brain_expression$Brain_expressed != 'No']
idx <- which(colnames(genexp) %in% brain_genes)
genexp <- genexp[,idx] # 1340 x 14370


#translate into synonyms
vec <- covarianceSelection::symbol_synonyms(colnames(genexp), verbose = T)
unknown_genes_idx <- which(is.na(vec))
vec <- vec[-unknown_genes_idx]
genexp <- genexp[-unknown_genes_idx] # 1340 x 14297
colnames(genexp) <- vec

#average non-unique genes
genexp <- covarianceSelection:::average_same_columns(genexp) # 1340 x 14246

#remove samples from subregions that we don't have a region for
region_subregion <- covarianceSelection::region_subregion # https://github.com/linnykos/covarianceSelection/blob/master/covarianceSelection/R/data.R
vec <- rownames(genexp)
# extract regions from colnames 
subregion <- unlist(strsplit(vec,"\\."))[seq(2, length(vec)*4, 4)] # split string separated by "." https://stackoverflow.com/questions/26665100/how-to-use-the-strsplit-function-with-a-period
idx <- which(subregion %in% region_subregion$subregion)
genexp <- genexp[idx,] # 1294 x 14249 --> 1294 x 14238

####### check how mmany top SCHEMA genes are missing in genexp 
#scz1$Gene[which(scz1$Gene[1:200] %in% colnames(genexp)==F)]


################
# explore the notations in row names 
#sample_names=rownames(genexp)
#comma_position=gregexpr(".", sample_names[1])
#sample_names[grepl("VFC",sample_names)] # find samples having "VFC"
################
```




```{r, echo=F, message=F, warning=F, results=F, eval=T}
multiplesheets <- function(fname) {
   
  # getting info about all excel sheets
  sheets <- readxl::excel_sheets(fname)
  tibble <- lapply(sheets, function(x) readxl::read_excel(fname, sheet = x))
  data_frame <- lapply(tibble, as.data.frame)
    
  # assigning names to data frames
  names(data_frame) <- sheets
    
  # print data frame
  print(data_frame)
}

####
# load SCZ gene 
SCHEMA_gene_results=multiplesheets("C:\\han\\Dataset\\SCHEMA\\41586_2022_4556_MOESM3_ESM.xlsx")
scz1=data.frame(Gene=SCHEMA_gene_results$`Table S5 - Gene Results`$`Gene Symbol`, pval.meta=SCHEMA_gene_results$`Table S5 - Gene Results`$`P meta`, qval.meta=SCHEMA_gene_results$`Table S5 - Gene Results`$`Q meta`, dn_PTV=SCHEMA_gene_results$`Table S5 - Gene Results`$`De novo PTV`) 
scz1$dn_PTV[is.na(scz1$dn_PTV)]=0   # fill missing PTV with 0 
scz1=scz1%>% drop_na()  # remove NA's  ######## this is important 

scz2=scz1[-c(9934,  9935,  9936, 13100),]   #  remove duplicates which(duplicated(scz1$Gene)==T)
#unique_gene= unique(scz1$Gene)
#scz=data.frame(Gene=unique_gene, pval.meta=scz1$pval.meta[which(scz1$Gene %in% unique_gene)])  # remove duplicates 
```




```{r, echo=F, message=F, warning=F, eval=F}
#match the order in both datasets ########## check this part 
idx <- which(colnames(genexp) %in% scz2$Gene)
genexp <- genexp[,idx] # 1294 x 13964
idx <- which(scz2$Gene %in% colnames(genexp))
scz3=scz2[idx,] # 4 risk genes not in gene expression data:  scz2$Gene[which(!scz2$Gene[1:32]%in%colnames(genexp))]



idx <- covarianceSelection::matching(scz3$Gene, colnames(genexp))
genexp <- genexp[,idx]  # 1294 x 12879 

dat_list <- covarianceSelection::extractor(genexp) # 212 partitions
dat_list <- lapply(dat_list, as.matrix, drop = F)

# remove partitions too small
idx <- which(sapply(dat_list, function(x){ifelse(nrow(x) >= 5, T, F)}))
dat_list <- dat_list[idx] # 125 partitions
dat_list <- lapply(dat_list, scale, scale = F)  # 125 partitions 

#if(verbose) print(paste0("Dimension of genexp is: ", paste0(dim(genexp), collapse = ", ")))

#cleanup
#rm(list = c("brain_expression", "brain_genes", "idx", "vec", "region_subregion",
#            "subregion", "genexp", "unknown_genes_idx"))

#save.image(file = paste0(save_filepath, "/step0_loading", filepath_suffix, ".RData"))

```




```{r, echo=F, message=F, warning=F, eval=F}
# step1_screening.R screens the genes according to Liu et al. (2015). This is reported in Section 6.1.
# https://github.com/linnykos/covarianceSelection/blob/master/main/step1_screening.R
p_thres_screening <- 0.01 
p_primary <- 0.1
num_genes <- 3100 # constraint genes with pLI>0.9 https://www.ncbi.nlm.nih.gov/pmc/articles/PMC9805802/
verbose=T

###

#if(verbose) print(paste0(Sys.time(), "Start of step 1: Screening"))

selected_idx <- grep("PFC\\.[3-5]", names(dat_list)) # [3-5] should be early PC in Liu et.al 2015 (10-19 post conception week) 
dat_pfc35 <- do.call(rbind, dat_list[selected_idx]) # 107 x 13964  --> 107 x 12879  


screening_res <- covarianceSelection::screen(dat_pfc35, pv = scz3$pval.meta, p_thres = p_thres_screening, 
                                             num_genes = num_genes)

# 193 primary, 3307 secondary, total of 3500, they rank correlations between primary genes and the rest to pick the top genes to have 3500 genes in total, don't use correlation threshold tau in Algorithm 1 of AOAS 2015

# reorder which genes are primary and which are secondary
all_idx <- sort(unique(c(screening_res$primary, screening_res$secondary)))
screening_res$primary <- all_idx[which(scz3$pval.meta[all_idx] < p_primary)]
screening_res$secondary <- setdiff(all_idx, screening_res$primary)

# apply the new gene list
for(i in 1:length(dat_list)){
  dat_list[[i]] <- dat_list[[i]][,c(screening_res$primary, screening_res$secondary)]
}

 scz=scz3[c(screening_res$primary, screening_res$secondary),]   # 3500 x 4 
                
#if(verbose) print(paste0("Dimension of dat_list is: ", unique(sapply(dat_list, ncol)), collapse = ", "))

#rm(list = c("selected_idx", "dat_pfc35", "i"))

#save.image(file = paste0(save_filepath, "/step1_screening", filepath_suffix, ".RData"))
# save(dat_pfc35, scz, file="../data/dat_pfc35_20240328.RData")
```



```{r, echo=F, message=F, warning=F, eval=F}
# https://github.com/linnykos/covarianceSelection/blob/master/main/step2_nodawn_analysis.R
# step2_nodawn_analysis.R detects the risk genes only based on the TADA dataset.

fdr_cutoff_alt <- 0.05
verbose=T

if(verbose) print(paste0(Sys.time(), "Start of step 2: No DAWN analysis"))

genes_nodawn <- sort(as.character(scz[which(scz$qval.meta <= fdr_cutoff_alt),"Gene"]))
length(genes_nodawn)

#save.image(file = paste0(save_filepath, "/step2_nodawn_analysis", filepath_suffix, ".RData"))
```


```{r, echo=F, message=F, warning=F, eval=F}
# step3_pfc35_analysis.R implicates risk genes in the DAWN framework using the Window 1B partitions. This is reported in Section 6.4.
# https://github.com/linnykos/covarianceSelection/blob/master/main/step3_pfc35_analysis.R
fdr_cutoff <- 0.05
pthres <- 0.05
verbose=T

#####
set.seed(10)
if(verbose) print(paste0(Sys.time(), "Start of step 3: Naive analysis"))

selected_idx <- grep("PFC\\.[3-5]", names(dat_list))
dat_pfc35 <- do.call(rbind, dat_list[selected_idx]) # 107 x 3065
dat_pfc35 <- scale(dat_pfc35, scale = F)
```


```{r, echo=F, message=F, warning=F, eval=F}
load("../data/dat_pfc35_20240328.RData")
# estimate graphical model on PFC35 using cross-validated lasso for neighborhood selection
 res <- covarianceSelection::graphicalModel_range(dat_pfc35, 1:length(screening_res$primary), 
                                                  lambda_min = 0.01, lambda_max = 0.35, 
                                                  lambda_length = 30, verbose = T) 
# save.image(file = paste0(save_filepath, "/step3_pfc35_analysis", filepath_suffix, ".RData"))
# 
 scale_vec_pfc35 <- sapply(res, function(x){covarianceSelection::compute_scale_free(as.matrix(x$adj_mat))}) # equation  3.2 of AOAS paper 
# edges_vec_pfc35 <- sapply(res, function(x){sum(as.matrix(x$adj_mat))/2})
 idx <- which.max(scale_vec_pfc35) # choose tuning parameter lambda maximizing R^2 
# idx <- 27
 adj_pfc35 <- as.matrix(res[[idx]]$adj_mat)



#stopifnot(all(dim(adj_pfc35) == nrow(tada)))
stopifnot(all(dim(adj_pfc35) == nrow(scz)))

# run the HMRF
set.seed(10)
seedindex <- rep(0, ncol(adj_pfc35))
#seedindex[which(tada$dn.LoF >= 3)] <- 1  ### seedindex: a (0,1) vector of length d, where 1 means the gene is in group 1
seedindex[which(scz$dn_PTV >= 1)] <- 1

if(verbose) print(paste0(Sys.time(), ": HMRF"))
set.seed(10)

scz$pval.meta[which(scz$pval.meta==1)]=1-10^{-7}; # enforce 1 pvalues to be slightly less than 1  
hmrf_pfc35 <- covarianceSelection::hmrf(scz$pval.meta, adj_pfc35, seedindex, pthres = 0.1)  # pval must not be 0 or 1's, i.e between 0 and 1  
report_pfc35 <- covarianceSelection::report_results(scz$Gene, 1-hmrf_pfc35$post, scz$pval.meta, hmrf_pfc35$Iupdate)
genes_pfc35 <- sort(as.character(report_pfc35$Gene[which(report_pfc35$FDR <= fdr_cutoff)]))
#scz2$Gene[which(!scz2$Gene[1:32]%in%genes_pfc35)]  # of 32 risk genes, how many are missed by DAWN
#scz3$Gene[which(!scz3$Gene[1:28]%in%genes_pfc35)]
#new_genes_pfc35=setdiff(genes_pfc35,scz3$Gene[1:28])

adj_pfc35 <- Matrix::Matrix(adj_pfc35, sparse = T)
#save(genes_pfc35, adj_pfc35, file="../data/gene_pfc35_20240416.RData")
```


## Enrichment of risk genes 

### overlap with SCHEMA risk genes


```{r, echo=F, message=F, warning=F}
load("C:/han/Projects/Multi_Omics/multi_omics/data/gene_pfc35_20240416.RData")

SCHEMA_top_10_gene=scz1$Gene[1:10]
SCHEMA_FDR_gene=scz1$Gene[1:32]

X=list(
SCHEMA_10_Risk_Gene=SCHEMA_top_10_gene,
DAWN_gene=genes_pfc35,
SCHEMA_FDR_gene=SCHEMA_FDR_gene
  )


#library(ggvenn)
#ggvenn(
#  X, 
#  fill_color = c("#0073C2FF", "#EFC000FF",  "#CD534CFF"),
#  stroke_size = 0.5, set_name_size = 4
#  )

#png("C:\\Users\\hans\\OneDrive - Marquette University\\SCHEMA\\Fig_venn_diagram.png")

library("VennDiagram") 
# move to new plotting page 
grid.newpage() 
  
# create Venn diagram with three sets 
draw.triple.venn(area1=10,area2=46, area3 =32,  
                 n12=9, n23=28, n13=10, n123=9,  
                 category=c("SCHEMA 10 risk genes","DAWN genes","SCHEMA FDR genes"), 
                 col="red",fill=c("Green","Orange","Blue"), lty="solid", 
                 cat.cex = 1.3, cat.pos=0, cat.col=c("Green","Orange","Blue"), cex=1.5) # cat.cex  the size of the category names; cex: A vector (length 7) giving the size of the areas' labels; cat.pos: position of category names https://www.rdocumentation.org/packages/VennDiagram/versions/1.7.3/topics/draw.triple.venn
#dev.off()
```

* 4 SCHEMA FDR genes are missed by DAWN is because these 4 genes don't have expression data in brain span. 




### enrichment with other gene sets 

### GO enrichment analsysis 




```{r, echo=F, message=F, warning=F, eval=F}

############# use igraph to draw graph
#library(igraph)
#network <- graph_from_incidence_matrix(adj_pfc35)
#plot(network)



library(statnet)
#adjacency_network=adjacency_strong[-row_sub, -row_sub]
#Gene_selected=colnames(adjacency)[-row_sub]
adjacency_network=adj_pfc35
Gene_selected=genes_pfc35
rownames(adjacency_network)=seq(1:nrow(adjacency_network))
colnames(adjacency_network)=seq(1:ncol(adjacency_network))
dim(adjacency_network)
net=network(as.matrix(adjacency_network), directed=FALSE) # https://web.stanford.edu/class/bios221/labs/networks/lab_7_networks.html
#nodeColors<-ifelse(Gene_selected %in% Gene_selected,"red","red")
#plot(net, label=Gene_selected,vertex.col=nodeColors, vertex.size=10, label.cex=0.5) # https://statnet.org/workshop-intro-sna-tools/
plot(net, label=seq(1:ncol(adjacency_network)), vertex.size=10, label.cex=0.5) # https://statnet.org/workshop-intro-sna-tools/


#scz1[which(scz1$Gene %in% genes_pfc35),]
# fdr005_gene=sort(as.character(report_pfc35$Gene[which(report_pfc35$FDR <= 0.05)]))
# scz1[which(scz1$Gene %in% fdr005_gene),]


#rm(list = c("dat_pfc35", "seedindex", "idx", "res", "selected_idx"))

#save.image(file = paste0(save_filepath, "/step3_pfc35_analysis", filepath_suffix, ".RData"))

########################

 #plot(exp(seq(log(0.01), log(0.35), length.out = length(scale_vec_pfc35))), scale_vec_pfc35)
```



```{r echo=F, message=F, warning=F, eval=F}
# step3_pfc35_analysis.R implicates risk genes in the DAWN framework using the Window 1B partitions. This is reported in Section 6.4.
# https://github.com/linnykos/covarianceSelection/blob/master/main/step3_pfc35_analysis.R
fdr_cutoff <- 0.05
pthres <- 0.05
verbose=T

#####
set.seed(10)
if(verbose) print(paste0(Sys.time(), "Start of step 3: Naive analysis"))

selected_idx <- grep("PFC", names(dat_list))
dat_pfc <- do.call(rbind, dat_list[selected_idx]) # 107 x 3065
dat_pfc <- scale(dat_pfc, scale = F)

# estimate graphical model on PFC35 using cross-validated lasso for neighborhood selection
# res <- covarianceSelection::graphicalModel_range(dat_pfc, 1:length(screening_res$primary), 
#                                                  lambda_min = 0.01, lambda_max = 0.35, 
#                                                  lambda_length = 30, verbose = T) 
# save.image(file = paste0(save_filepath, "/step3_pfc35_analysis", filepath_suffix, ".RData"))
# 
# scale_vec_pfc <- sapply(res, function(x){covarianceSelection::compute_scale_free(as.matrix(x$adj_mat))})
# edges_vec_pfc35 <- sapply(res, function(x){sum(as.matrix(x$adj_mat))/2})
# # idx <- which.max(scale_vec_pfc35)
# idx <- 27
# adj_pfc35 <- as.matrix(res[[idx]]$adj_mat)

res <- covarianceSelection::graphicalModel(dat_pfc, primary_idx = 1:length(screening_res$primary), lambda = seq(0.05, 0.1, length.out = 15)[1])
adj_pfc <- as.matrix(res$adj_mat)


#stopifnot(all(dim(adj_pfc35) == nrow(tada)))
stopifnot(all(dim(adj_pfc35) == nrow(scz)))

# run the HMRF
set.seed(10)
seedindex <- rep(0, ncol(adj_pfc))
#seedindex[which(tada$dn.LoF >= 3)] <- 1  ### seedindex: a (0,1) vector of length d, where 1 means the gene is in group 1
seedindex[which(scz$dn_PTV >= 1)] <- 1

if(verbose) print(paste0(Sys.time(), ": HMRF"))
set.seed(10)
#hmrf_pfc35 <- covarianceSelection::hmrf(tada$pval.TADA, adj_pfc35, seedindex, pthres = pthres)
#report_pfc35 <- covarianceSelection::report_results(tada$Gene, 1-hmrf_pfc35$post, tada$pval.TADA, hmrf_pfc35$Iupdate)
#genes_pfc35 <- sort(as.character(report_pfc35$Gene[which(report_pfc35$FDR <= fdr_cutoff)]))

scz$pval.meta[which(scz$pval.meta==1)]=1-10^{-7}; # enforce 1 pvalues to be slightly less than 1  
hmrf_pfc <- covarianceSelection::hmrf(scz$pval.meta, adj_pfc, seedindex, pthres = 0.1)  # pval must not be 0 or 1's, i.e between 0 and 1  

report_pfc <- covarianceSelection::report_results(scz$Gene, 1-hmrf_pfc$post, scz$pval.meta, hmrf_pfc$Iupdate)
genes_pfc <- sort(as.character(report_pfc$Gene[which(report_pfc$FDR <= fdr_cutoff)]))
#scz2$Gene[which(!scz2$Gene[1:32]%in%genes_pfc35)]  # of 32 risk genes, how many are missed by DAWN
#scz3$Gene[which(!scz3$Gene[1:28]%in%genes_pfc)]

adj_pfc <- Matrix::Matrix(adj_pfc, sparse = T)

#library(igraph)
#network <- graph_from_incidence_matrix(adj_pfc35)
#plot(network)

#scz1[which(scz1$Gene %in% genes_pfc35),]
# fdr005_gene=sort(as.character(report_pfc35$Gene[which(report_pfc35$FDR <= 0.05)]))
# scz1[which(scz1$Gene %in% fdr005_gene),]


#rm(list = c("dat_pfc35", "seedindex", "idx", "res", "selected_idx"))

#save.image(file = paste0(save_filepath, "/step3_pfc35_analysis", filepath_suffix, ".RData"))

########################

 #plot(exp(seq(log(0.01), log(0.35), length.out = length(scale_vec_pfc35))), scale_vec_pfc35)
```




```{r, echo=F, message=F, warning=F, eval=F}
#png("../figures/figure_9.png", height = 1000, width = 2000, units = "px", res = 300)
par(mar = c(0,0,3,0), mfrow = c(1,2))
set.seed(10)
igraph::plot.igraph(g_selected, vertex.label = NA, main = "Full graph")

par(mar = c(3,3,3,0.5))
# next plot the adjacency matrix
image(.rotate(adj_tmp), asp = T, col = color_palatte, breaks = c(-.5,.5,1.5), xaxt = "n", yaxt = "n",
      xlab = "", ylab = "", main = "Adjacency matrix (subgraph)", axes = F)
title(ylab = "Index locations", mgp = c(1,1,0))
title(xlab = "Index locations", mgp = c(1,1,0))

# put in dashed lines
x_width <- length(idx_our)/nrow(adj_tmp)
y_height <- 1 - x_width
lines(rep(x_width, 2), c(1,1-x_width), lwd = 2, lty = 2)
lines(c(0,x_width), rep(y_height, 2), lwd = 2, lty = 2)
#graphics.off()
```




## gene co-expression by WGCNA  

* [reference1](https://fuzzyatelin.github.io/bioanth-stats/module-F21-Group1/module-F21-Group1.html)
* [reference2](https://rstudio-pubs-static.s3.amazonaws.com/687551_ed469310d8ea4652991a2e850b0018de.html)
* [reference3](https://alexslemonade.github.io/refinebio-examples/04-advanced-topics/network-analysis_rnaseq_01_wgcna.html)


```{r, echo=F, message=F, warning=F}
#load("../data/dat_pfc35_20240328.RData")
load("C:/han/Projects/Multi_Omics/multi_omics/data/dat_pfc35_20240328.RData")
dat_pfc35_constraint_gene=dat_pfc35[,which(colnames(dat_pfc35) %in% scz$Gene)] # extract gene expression for 3100 genes at PFC
gsg=goodSamplesGenes(dat_pfc35_constraint_gene) # identify outlier genes 
summary(gsg) # no outlier genes 
```

```{r, echo=F, message=F, warning=F}
### identify outlier samples https://fuzzyatelin.github.io/bioanth-stats/module-F21-Group1/module-F21-Group1.html#Identifying_outlier_samples
sampleTree <- hclust(dist(dat_pfc35_constraint_gene), method = "average") #Clustering samples based on distance 

#Setting the graphical parameters
par(cex = 0.6);
par(mar = c(0,4,2,0))

#Plotting the cluster dendrogram
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)



par(cex = 0.6);
par(mar = c(0,4,2,0))
plot(sampleTree, main = "Sample clustering to detect outliers", sub="", xlab="", cex.lab = 1.5,
cex.axis = 1.5, cex.main = 2)
#draw on line to show cutoff height
abline(h = 40, col = "red")


cut.sampleTree <- cutreeStatic(sampleTree, cutHeight = 40, minSize = 10) #returns numeric vector
#Remove outlier
expression.data <- dat_pfc35_constraint_gene[cut.sampleTree==1, ]
dim(expression.data)
```
```{r, echo=F, message=F, warning=F}
spt <- pickSoftThreshold(expression.data)
#spt

#par(mar=c(1,1,1,1))
plot(spt$fitIndices[,1],spt$fitIndices[,2],
xlab="Soft Threshold (power)",ylab="Scale Free Topology Model Fit,signed R^2",type="n",
main = paste("Scale independence"))
text(spt$fitIndices[,1],spt$fitIndices[,2],col="red")
abline(h=0.80,col="red")

#par(mar=c(1,1,1,1))
plot(spt$fitIndices[,1], spt$fitIndices[,5],
xlab="Soft Threshold (power)",ylab="Mean Connectivity", type="n",
main = paste("Mean connectivity"))
text(spt$fitIndices[,1], spt$fitIndices[,5], labels= spt$fitIndices[,1],col="red")
```

* choose $\beta$ that maximizes $R^2$, and minimizes mean connectivity 



```{r, echo=F, message=F, warning=F}
softPower <- 8
adjacency <- adjacency(expression.data, power = softPower)
#dim(adjacency)
threshold=0.5
adjacency_strong=matrix(nrow=nrow(adjacency), ncol=ncol(adjacency))
k=0; row_sub=numeric()
for(i in 1:nrow(adjacency))
{
  for (j in 1:ncol(adjacency))
  {
    if (abs(adjacency[i,j])<=threshold)
         adjacency_strong[i,j]=0
     if (abs(adjacency[i,j])>threshold)
    adjacency_strong[i,j]=1
  } 
  if (sum(adjacency_strong[i,])==1)
  {
    k=k+1
    row_sub[k]=i
  }
}  # end of for(i in 1:nrow(adjacency))
library(statnet)
adjacency_network=adjacency_strong[-row_sub, -row_sub]
Gene_selected=colnames(adjacency)[-row_sub]
rownames(adjacency_network)=Gene_selected
colnames(adjacency_network)=Gene_selected
dim(adjacency_network)
net=network(adjacency_network, directed=FALSE) # https://web.stanford.edu/class/bios221/labs/networks/lab_7_networks.html
nodeColors<-ifelse(Gene_selected %in% Gene_selected,"red","red")
plot(net, label=Gene_selected,vertex.col=nodeColors, vertex.size=10, label.cex=0.5) # https://statnet.org/workshop-intro-sna-tools/



#### overlap between 32 FDR genes and co-expressed genes 
overlap_gene=intersect(scz[scz$qval.meta<0.05,]$Gene, Gene_selected)


nodeColors<-ifelse(Gene_selected%in%overlap_gene,"yellow","red")
plot(net,label=Gene_selected,vertex.col=nodeColors,  vertex.size=10, label.cex=0.5)

#install.packages("statnet")
#library(statnet)
#heatmap(adjacency)
#plot(graph_from_adjacency_matrix(adjacency))
```




```{r, echo=F, message=F, warning=F}
#https://fuzzyatelin.github.io/bioanth-stats/module-F21-Group1/module-F21-Group1.html#Network_Construction
TOM <- TOMsimilarity(adjacency)
TOM.dissimilarity <- 1-TOM
#creating the dendrogram 
geneTree <- hclust(as.dist(TOM.dissimilarity), method = "average")
#plotting the dendrogram
#sizeGrWindow(12,9)
plot(geneTree, xlab="", sub="", main = "Gene clustering on TOM-based dissimilarity", 
labels = FALSE, hang = 0.04)


Modules <- cutreeDynamic(dendro = geneTree, distM = TOM.dissimilarity, deepSplit = 2, pamRespectsDendro = FALSE, minClusterSize = 30)
table(Modules) #returns a table of the counts of factor levels in an object. In this case how many genes are assigned to each created module.

ModuleColors <- labels2colors(Modules) #assigns each module number a color
table(ModuleColors) #returns the counts for each color (aka the number of genes within each module)

#plots the gene dendrogram with the module colors
plotDendroAndColors(geneTree, ModuleColors,"Module",
dendroLabels = FALSE, hang = 0.03,
addGuide = TRUE, guideHang = 0.05,
main = "Gene dendrogram and module colors")
```

